frontend/app/src/store/balances/index.ts

```diff
@@ -28,15 +31,32 @@ export const useBalancesStore = defineStore('balances', () => {
   const { queryBalancesAsync } = useBalancesApi();
   const priceStore = useBalancePricesStore();
   const { prices } = storeToRefs(priceStore);
-  const { getAssetPrice, fetchPrices, fetchExchangeRates } = priceStore;
+  const { getAssetPrice, fetchPrices, fetchExchangeRates, exchangeRate } =
+    priceStore;
   const { notify } = useNotifications();
   const { isTaskRunning, addTask } = useTasks();
   const { tc } = useI18n();
+  const { currencySymbol, currency } = storeToRefs(useGeneralSettingsStore());
 
   const adjustPrices = (prices: MaybeRef<AssetPrices>) => {
-    updateChainPrices(prices);
-    updateManualPrices(prices);
-    updateExchangePrices(prices);
+    const pricesConvertedToUsd = { ...get(prices) };
+
+    const mainCurrency = get(currencySymbol);
+    if (mainCurrency !== CURRENCY_USD) {
+      const rate = get(exchangeRate(mainCurrency)) ?? One;
+
+      for (const asset in pricesConvertedToUsd) {
+        const price = pricesConvertedToUsd[asset];
+
+        if (price.isCurrentCurrency) {
+          price.usdPrice = price.value.dividedBy(rate);
+        }
+      }
+    }
+
+    updateChainPrices(pricesConvertedToUsd);
+    updateManualPrices(pricesConvertedToUsd);
+    updateExchangePrices(pricesConvertedToUsd);
```

Comment:
I know this is temporary, but can you add some unit tests for this logic here.

Reply:
Unit test is added.